lib/membrane_rtc_engine/engine.ex

```diff
@@ -2,17 +2,111 @@ defmodule Membrane.RTC.Engine do
   @moduledoc """
   RTC Engine implementation.
 
-  One RTC Engine instance is responsible for managing one room in which
-  all tracks of one peer are forwarded to all other peers.
+  RTC Engine is an abstraction layer responsible for linking together different types of `Endpoints`.
+  From implementation point of view RTC Engine is a `Membrane.Pipeline`.
+
+  ## Endpoints
+
+  `Endpoints` are `Membrane.Bin`s able to publish own tracks and subscribe for tracks from other Endpoints.
+  One can thought about Endpoint as an entity responsible for handling some specifc task.
+  Endpoint can be added and removed using `add_endpoint/3` and `remove_endpoint/1` respectively.
+
+  There are basically two types of Endpoints:
+  * Standalone Endpoints - they are in most cases spawned only once per RTC Engine instance and they are not associated with any peer.
+  * Peer Endpoints - they are associated with some peer.
+  Associating Endpoint with Peer will cause RTC Engine sending some Media Events to client library
+  e.g. one which indicates which tracks belong to which peer.
+
+  Currently RTC Engine ships with implementation of two Endpoints:
+  * `#{inspect(__MODULE__)}.Endpoint.WebRTC` which is responsible for establishing connection with some WebRTC
+  peer (mainly browser) and exchanging media with it. WebRTC Endpoint is a Peer Endpoint.
+  * `#{inspect(__MODULE__)}.Endpoint.HLS` which is responsible for receiving media tracks from all other Endpoints and
+  saving them to a file by creating HLS playlist. HLS Endpoint is a Standalone Endpoint.
+
+  User can also implement custom Endpoints.
+
+  #### Implementing custom RTC Engine Endpoint
+
+  Each RTC Engine Endpoint has to:
+  * implement `Membrane.Bin` behavior
+  * specify input, output or both input and output pads depnding on what it is intended to do.
+    For example, if Endpoint will not publish any tracks but only subscribe for tracks from other Endpoints it can specify only input pads.
+    Pads should have following form
+
+    ```elixir
+      def_input_pad :input,
+        demand_unit: :buffers,
+        caps: :any,
+        availability: :on_request
+
+      def_output_pad :output,
+        demand_unit: :buffers,
+        caps: :any,
+        availability: :on_request
+    ```
+
+  * publish or subscribe for some tracks using `publish/1` or `subscribe/1` respectively.
+    `publish/1` returns some actions which have to be returend from Membrane callback.
+    Those actions will cause RTC Engine to send message in form of `{:new_tracks, tracks}`
```

Comment:
To publish tracks user has to return notification action from Membrane callback. This notification action will be fetched by RTC Engine and it will then publish info about tracks to all other peers.
To make it easier we have created function called `publish` that will create notification action for the user.

Reply:
Ok, sorry I misread that